### PR TITLE
Add subpanel header section & close action

### PR DESCRIPTION
**Pull request by iisa**
Adds header section to the open subpanel.
The header takes `{ title, menuDetails, actionButton } = menuObject` and uses it to draw the subpanel's header.
Has:
- new menu params: `{ title, menuDetails, actionButton }`
  - allows for parent component to connect dynamic values to `menuDetails` section
    - ex. always updating header to reflect user's search results total
  - `menuDetails` also wraps under main title if it overflows
  - adds for extra action button that the parent component can set inside of the submenu using `actionButton` parameter
- has new close button & emits custom event on close
  - this closes loop on the item navigator's hot key functionality
- add accent color to open subpanel

### To test:
-  pull down branch
- run command `npm install && npm start`
- local demo starts
- open menu -> first example has the new header attributes
- click of extra action button will fire a console.log in the dev console

ezgif com-video-to-gif (4)

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -123,49 +128,70 @@ <h1>${this.title}</h1>
...
 
-    document.querySelector('section').appendChild(menuSlider);
-
-    document.querySelector('button').addEventListener('click', () => {
-      menuSlider.open = !menuSlider.open;
-    });
+    (function toggleScope() {
+      let menuOpen = false;
+      const closeCB = (event) => {
+        console.log('Drawer Closed', event);
```
Comment (bfalling) [inline demo/index.html:182]: QUESTION: Did you want this debug info in here?

File: src/ia-menu-slider.js [hunk 2]
```diff
@@ -59,13 +65,50 @@ export class IAMenuSlider extends LitElement {
     return this.selectedMenu ? 'open' : 'closed';
   }
 
+  /**
+   * closes menu drawer
+   */
+  closeMenu() {
+    this.open = false;
+    const drawerClosed = new CustomEvent('ItemNavMenuClosed', {
```
Comment (bfalling) [inline src/ia-menu-slider.js:73]: QUESTION: I sometimes think about the risks of using free strings here... so easy to misspell. Should we be using constants instead? (Allow for static checking too!)

File: src/ia-menu-slider.js [hunk 3]
```diff
@@ -59,13 +65,50 @@ export class IAMenuSlider extends LitElement {
...
+      detail: this.selectedMenuDetails,
+    });
+    this.dispatchEvent(drawerClosed);
+  }
+
+  /**
+   * renders menu header
+   */
+  get renderMenuHeader() {
+    // const { selectedMenu } = this;
```
Comment (bfalling) [inline src/ia-menu-slider.js:83]: SUGGESTION: Remove comment?

File: src/ia-menu-slider.js [hunk 4]
```diff
@@ -59,13 +65,50 @@ export class IAMenuSlider extends LitElement {
...
+    const { title = '', menuDetails = '', actionButton } = this.selectedMenuDetails || {};
+    const actionSection = actionButton
+      ? html`<div class="custom-action">${actionButton}</div>`
+      : nothing;
+
+    return html`
+      <header>
+        <div class='details'>
```
Comment (bfalling) [inline src/ia-menu-slider.js:91]: SUGGESTION: Purely cosmetic, but in keeping consistent with your convention below, maybe want to use double-quotes for this?

File: test/ia-menu-slider.test.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { IAMenuSlider } from '../src/ia-menu-slider.js';
 customElements.define('ia-menu-slider', IAMenuSlider);
 
 const container = (menus = []) => (
-  html`<ia-menu-slider .menus=${menus}></ia-menu-slider>`
+  html`<ia-menu-slider .menus=${menus} @ItemNavMenuClosed=${() => {}}></ia-menu-slider>`
```
Comment (bfalling) [inline test/ia-menu-slider.test.js:7]: QUESTION: Did you want to be testing `<ia-menu-slider>` with its Boolean `open` attribute set in some way too?

File: test/ia-menu-slider.test.js [hunk 6]
```diff
@@ -138,4 +138,81 @@ describe('<ia-menu-slider>', () => {
...
+    const menu = {
+      icon: html`<ia-icon icon="audio"></ia-icon>`,
+      title: 'Audio Menu',
+      menuDetails: '(100,000,000 tracks)',
+      actionButton: html`
+        <button
+          style='color: white;'
+          @click=${extraActionClickHandler}
+        >Foo
+        </button>
```
Comment (bfalling) [inline test/ia-menu-slider.test.js:156]: SUGGESTION/QUESTION: If you're going to make sure to put `Foo` on the line with the `>` to make sure it has no whitespace before `Foo`, then perhaps you want to put `</button>` directly after `Foo` so there's also no whitespace after `Foo`...?

**Review by bfalling — APPROVED** [review]
A few questions and suggestions. No blockers.

File: demo/index.html (hunk 1 above)
Comment (iisa) [reply demo/index.html:182]: hi yes, thanks for asking. it's for the demo.

File: src/ia-menu-slider.js (hunk 4 above)
Comment (iisa) [reply src/ia-menu-slider.js:91]: thanks for catching.

File: test/ia-menu-slider.test.js (hunk 6 above)
Comment (iisa) [reply test/ia-menu-slider.test.js:156]: thanks, moved it up.

File: src/ia-menu-slider.js (hunk 2 above)
Comment (iisa) [reply src/ia-menu-slider.js]: I've moved it up to its own object so we can use a reference in the code.
I'll think about it further and consult our lit element team to see what is the best approach on an ecosystem level as there are other places that this event name is being referenced.

File: test/ia-menu-slider.test.js (hunk 5 above)
Comment (iisa) [reply test/ia-menu-slider.test.js:7]: Good question. We are actually testing the `open` boolean in various unit tests using the class/element instance.  so we are covered in accessing and changing that property.

I've actually updated the demo to account for toggling `this.open` from the outside.

This is actually an artifact from devving, as well.  yanked.

**Pull request merged**